go/types/simplify.go

```diff
@@ -0,0 +1,225 @@
+package types
+
+import (
+	"github.com/attic-labs/noms/go/d"
+	flag "github.com/juju/gnuflag"
+)
+
+var enableTypeSimplification = false
+
+func RegisterTypeSimplificationFlags(flags *flag.FlagSet) {
+	flags.BoolVar(&enableTypeSimplification, "type-simplificatino", false, "enables type simplification (see https://github.com/attic-labs/noms/issues/2995)")
+}
+
+func accreteTypes(t ...*Type) *Type {
+	if enableTypeSimplification {
+		return makeSimplifiedUnion(t...)
+	} else {
+		return MakeUnionType(t...)
+	}
+}
+
+// makeSimplifiedUnion returns a type that is a supertype of all the input types, but is much
```

Comment:
You are right. I thought for so long making sure I got it right and still I failed.